### ISSUE TITLE
`(int)` cast of `double` rounds the double instead of truncating it

### ISSUE THREAD

**Issue by Crown0815** [codingseb/ExpressionEvaluator#130]
In `C#` the call

```
var x = (int) 3.6;
```
produces `x = 3`.

The same call in ExpressionEvaluator produces `x = 4` instead.

This is due to the usage of 

```
...
return Convert.ChangeType(value, conversionType);
```

[in line 4150](https://github.com/codingseb/ExpressionEvaluator/blob/master/CodingSeb.ExpressionEvaluator/ExpressionEvaluator.cs#L4150), which for the given example resolves to `Convert.ChangeType(3.6, typeof(int))` which [indeed resolves to 4](https://dotnetfiddle.net/MxmoCo).

I understand if this is a design decision, but it might be a source of issues. We use ExpressionEvaluator to simulate `C#` code, which means for us, it is non-expected behavior.

For our intents and purposes it would be great to have an option to turn integer casting from `round` to `truncate` e.g. with 

```
evaluator.OptionTruncateOnIntegerCast = true;
```

This would preserve current behavior but enable a more correct simulation of `C#` behavior.

**Comment by codingseb** [comment]
Hello @Crown0815 and thank you for this issue.
You are right it should return 3.
I think there is no need for an option for this as ExpressionEvaluator should behave as C#. 
For me it's more a bug to correct.
I thought that Convert was working like a cast but I need to make a bit more test around this.

**Comment by codingseb** [comment]
I published version 1.4.37.0 
That correct this bug

**Comment by Crown0815** [comment]
@codingseb Amazing. I just tested it and it works as expected. Thank you so much for taking care of this in such a short timeframe.
